File: app/Exceptions/Upstream/UpstreamServerException.php
<?php

namespace Irail\Exceptions\Upstream;

use Irail\Exceptions\IrailHttpException;
use Throwable;

class UpstreamServerException extends IrailHttpException
{
    function __construct(string $message = '', int $code = 502, ?Throwable $previous = null)
    {
        parent::__construct($code, $message, $previous);
    }
}

File: src/main/java/be/irail/api/exception/upstream/UpstreamServerException.java
package be.irail.api.exception.upstream;

import be.irail.api.exception.IrailHttpException;

public class UpstreamServerException extends IrailHttpException {
    public UpstreamServerException(String message) {
        super(504, "An upstream server encountered an issue while handling this request: " + message);
    }
    public UpstreamServerException(String message, Exception e) {
        super(504, "An upstream server encountered an issue while handling this request: " + message, e);
    }

}
